### ISSUE TITLE
Add Support for Query String/Parameter Templating

### ISSUE THREAD

**Issue by spaceraccoon** [projectdiscovery/nuclei#4]
It would be awesome to have query string support in your fuzzing templates, similar to https://github.com/ameenmaali/qsfuzz. For example, the template would look like:

```
requests:
  - method: GET
    path:
      - "{{BaseURL}}"
    parameters:
      - key1=value1
      - key2=value2
    matchers:
      - type: word
        words:
          - "Vulnerable"
```

Nuclei would then look for `key` in `BaseURL` and replace its value with `value1`. If `key` doesn't exist, it'll append it to the URL. There also needs to be some logic to check if the URL already has other query strings, in which case `key` will be appended as `&key1=value1` instead of `?key1=value1`.

Finally, there will need to be additional support for the `POST` equivalent as well as possibly other types of encodings `multipart/form-data`, `application/x-www-form-urlencoded`, `application/json`. 

I understand this is a big ask and would be happy to contribute!

**Comment by Ice3man543** [comment]
@spaceraccoon, i didn't understand the second case where you mention other content types etc like application/json multipart etc, could you give some examples to elaborate more on the issue?

**Comment by edermi** [comment]
I think this would be a really powerful addition to allow testing for weak/default passwords. I think this is also the reason for the request of different encodings, since APIs will probably speak JSON as regular web applications may rely on `multipart/form-data` (especially when used with loads of JS involved in the login) or `application/x-www-form-urlencoded`

**Comment by Mzack9999** [comment]
Implemented in #76 - Closing the issue
